{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "f_func <- \"./functions.ipynb\"\n",
    "eval(parse(text = system2('jupyter', c('nbconvert', f_func, '--to=script', '--stdout'), stdout = TRUE)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "DIR_IN=\"../../output/mapAspergillus/\"\n",
    "samples_metadata <- read.csv(file = \"../../output/samples_metadata.csv\") %>% as.data.table()\n",
    "list.files(DIR_IN, pattern = \"[0-9]_.*_R(1)(pos|neg)_TLEN_EndMotif.txt\") %>% tail()\n",
    "list.files(DIR_IN, pattern = \"___.*_R(1)(pos|neg)_TLEN_EndMotif.txt\") %>% tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (exists(\"Asp_IS_META\")){rm(\"Asp_IS_META\")}\n",
    "profvis({\n",
    "for (file in list.files(DIR_IN, pattern = \"1131492_.*_R(1|2)(pos|neg)_TLEN_EndMotif.txt\")){\n",
    "    print(file)\n",
    "    sample_tmp <- unlist(strsplit(file, \"_\"))[1] \n",
    "    taxID_tmp <- unlist(strsplit(file, \"_\"))[4]\n",
    "    mapGenome_tmp <- unlist(strsplit(file, \"_\"))[5]\n",
    "    side_tmp <- unlist(strsplit(file, \"_\"))[6]\n",
    "    \n",
    "    if(countLines(paste0(DIR_IN, file)) > 1){\n",
    "        pre_tmp_IS <- fread(paste0(DIR_IN, file), sep = \",\") %>% \n",
    "                    dplyr::rename(EndMotif_tmp = EndMotif) %>% \n",
    "                    mutate(sample_id = sample_tmp) %>% \n",
    "                    mutate(side = side_tmp) %>% \n",
    "                    mutate(taxID = taxID_tmp) %>% \n",
    "                    mutate(mapGenome = mapGenome_tmp) %>% \n",
    "                    mutate(read = substr(side_tmp, 0, 2)) %>% \n",
    "                    filter(TLEN != 0) %>% #remove TLEN not equal to 0\n",
    "                    filter(abs(TLEN) < 1000) %>% #filter max TLEN of 1000\n",
    "                    filter(!grepl(EndMotif_tmp, pattern = \"N\")) %>% #remove reads with end-motif including \"N\"\n",
    "                    group_by(TLEN, EndMotif_tmp, sample_id, side, taxID, mapGenome) %>% \n",
    "                    summarise(Count = sum(Count), .groups = \"keep\")\n",
    "        \n",
    "        if (grepl(side_tmp, pattern = \"neg\")){\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- as.character(reverseComplement(DNAStringSet(pre_tmp_IS$EndMotif_tmp))) \n",
    "        } else {\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- pre_tmp_IS$EndMotif_tmp\n",
    "        }\n",
    "        \n",
    "        tmp_IS_meta <- merge(x = samples_metadata, y = tmp_IS, by.x = \"sample_id\", by.y = \"sample_id\") \n",
    "        if (!exists(\"Asp_IS_META\")){Asp_IS_META <- tmp_IS_meta} else {\n",
    "            Asp_IS_META <- rbind(Asp_IS_META, tmp_IS_meta)}\n",
    "    }\n",
    "    if (exists(\"Asp_IS_META\")){\n",
    "        fwrite(Asp_IS_META, \"../../output/PP_figures/1131492_IS_meta.csv\", row.names = FALSE)\n",
    "    }\n",
    "}\n",
    "#})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (exists(\"Asp_IS_META\")){rm(\"Asp_IS_META\")}\n",
    "#profvis({\n",
    "for (file in list.files(DIR_IN, pattern = \"746128_.*_R(1|2)(pos|neg)_TLEN_EndMotif.txt\")){\n",
    "    print(file)\n",
    "    sample_tmp <- unlist(strsplit(file, \"_\"))[1] \n",
    "    taxID_tmp <- unlist(strsplit(file, \"_\"))[4]\n",
    "    mapGenome_tmp <- unlist(strsplit(file, \"_\"))[5]\n",
    "    side_tmp <- unlist(strsplit(file, \"_\"))[6]\n",
    "    \n",
    "    if(countLines(paste0(DIR_IN, file)) > 1){\n",
    "        pre_tmp_IS <- fread(paste0(DIR_IN, file), sep = \",\") %>% \n",
    "                    dplyr::rename(EndMotif_tmp = EndMotif) %>% \n",
    "                    mutate(sample_id = sample_tmp) %>% \n",
    "                    mutate(side = side_tmp) %>% \n",
    "                    mutate(taxID = taxID_tmp) %>% \n",
    "                    mutate(mapGenome = mapGenome_tmp) %>% \n",
    "                    mutate(read = substr(side_tmp, 0, 2)) %>% \n",
    "                    filter(TLEN != 0) %>% #remove TLEN not equal to 0\n",
    "                    filter(abs(TLEN) < 1000) %>% #filter max TLEN of 1000\n",
    "                    filter(!grepl(EndMotif_tmp, pattern = \"N\")) %>% #remove reads with end-motif including \"N\"\n",
    "                    group_by(TLEN, EndMotif_tmp, sample_id, side, taxID, mapGenome) %>% \n",
    "                    summarise(Count = sum(Count), .groups = \"keep\")\n",
    "        \n",
    "        if (grepl(side_tmp, pattern = \"neg\")){\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- as.character(reverseComplement(DNAStringSet(pre_tmp_IS$EndMotif_tmp))) \n",
    "        } else {\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- pre_tmp_IS$EndMotif_tmp\n",
    "        }\n",
    "        \n",
    "        tmp_IS_meta <- merge(x = samples_metadata, y = tmp_IS, by.x = \"sample_id\", by.y = \"sample_id\") \n",
    "        if (!exists(\"Asp_IS_META\")){Asp_IS_META <- tmp_IS_meta} else {\n",
    "            Asp_IS_META <- rbind(Asp_IS_META, tmp_IS_meta)}\n",
    "    }\n",
    "    if (exists(\"Asp_IS_META\")){\n",
    "        fwrite(Asp_IS_META, \"../../output/PP_figures/746128_IS_meta.csv\", row.names = FALSE)\n",
    "    }\n",
    "}\n",
    "#})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (exists(\"Asp_IS_META\")){rm(\"Asp_IS_META\")}\n",
    "#profvis({\n",
    "for (file in list.files(DIR_IN, pattern = \"5052_.*_R(1|2)(pos|neg)_TLEN_EndMotif.txt\")){\n",
    "    print(file)\n",
    "    sample_tmp <- unlist(strsplit(file, \"_\"))[1] \n",
    "    taxID_tmp <- unlist(strsplit(file, \"_\"))[4]\n",
    "    mapGenome_tmp <- unlist(strsplit(file, \"_\"))[5]\n",
    "    side_tmp <- unlist(strsplit(file, \"_\"))[6]\n",
    "    \n",
    "    if(countLines(paste0(DIR_IN, file)) > 1){\n",
    "        pre_tmp_IS <- fread(paste0(DIR_IN, file), sep = \",\") %>% \n",
    "                    dplyr::rename(EndMotif_tmp = EndMotif) %>% \n",
    "                    mutate(sample_id = sample_tmp) %>% \n",
    "                    mutate(side = side_tmp) %>% \n",
    "                    mutate(taxID = taxID_tmp) %>% \n",
    "                    mutate(mapGenome = mapGenome_tmp) %>% \n",
    "                    mutate(read = substr(side_tmp, 0, 2)) %>% \n",
    "                    filter(TLEN != 0) %>% #remove TLEN not equal to 0\n",
    "                    filter(abs(TLEN) < 1000) %>% #filter max TLEN of 1000\n",
    "                    filter(!grepl(EndMotif_tmp, pattern = \"N\")) %>% #remove reads with end-motif including \"N\"\n",
    "                    group_by(TLEN, EndMotif_tmp, sample_id, side, taxID, mapGenome) %>% \n",
    "                    summarise(Count = sum(Count), .groups = \"keep\")\n",
    "        \n",
    "        if (grepl(side_tmp, pattern = \"neg\")){\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- as.character(reverseComplement(DNAStringSet(pre_tmp_IS$EndMotif_tmp))) \n",
    "        } else {\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- pre_tmp_IS$EndMotif_tmp\n",
    "        }\n",
    "        \n",
    "        tmp_IS_meta <- merge(x = samples_metadata, y = tmp_IS, by.x = \"sample_id\", by.y = \"sample_id\") \n",
    "        if (!exists(\"Asp_IS_META\")){Asp_IS_META <- tmp_IS_meta} else {\n",
    "            Asp_IS_META <- rbind(Asp_IS_META, tmp_IS_meta)}\n",
    "    }\n",
    "    if (exists(\"Asp_IS_META\")){\n",
    "        fwrite(Asp_IS_META, \"../../output/PP_figures/5052_IS_meta.csv\", row.names = FALSE)\n",
    "    }\n",
    "}\n",
    "#})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (exists(\"Asp_IS_META\")){rm(\"Asp_IS_META\")}\n",
    "profvis({\n",
    "for (file in list.files(DIR_IN, pattern = \"[0-9]_.*_R(1|2)(pos|neg)_TLEN_EndMotif.txt\")){\n",
    "    print(file)\n",
    "    sample_tmp <- unlist(strsplit(file, \"_\"))[1] \n",
    "    taxID_tmp <- unlist(strsplit(file, \"_\"))[4]\n",
    "    mapGenome_tmp <- unlist(strsplit(file, \"_\"))[5]\n",
    "    side_tmp <- unlist(strsplit(file, \"_\"))[6]\n",
    "    \n",
    "    if(countLines(paste0(DIR_IN, file)) > 1){\n",
    "        pre_tmp_IS <- fread(paste0(DIR_IN, file), sep = \",\") %>% \n",
    "                    dplyr::rename(EndMotif_tmp = EndMotif) %>% \n",
    "                    mutate(sample_id = sample_tmp) %>% \n",
    "                    mutate(side = side_tmp) %>% \n",
    "                    mutate(taxID = taxID_tmp) %>% \n",
    "                    mutate(mapGenome = mapGenome_tmp) %>% \n",
    "                    mutate(read = substr(side_tmp, 0, 2)) %>% \n",
    "                    filter(TLEN != 0) %>% #remove TLEN not equal to 0\n",
    "                    filter(abs(TLEN) < 1000) %>% #filter max TLEN of 1000\n",
    "                    filter(!grepl(EndMotif_tmp, pattern = \"N\")) %>% #remove reads with end-motif including \"N\"\n",
    "                    group_by(TLEN, EndMotif_tmp, sample_id, side, taxID, mapGenome) %>% \n",
    "                    summarise(Count = sum(Count), .groups = \"keep\")\n",
    "        \n",
    "        if (grepl(side_tmp, pattern = \"neg\")){\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- as.character(reverseComplement(DNAStringSet(pre_tmp_IS$EndMotif_tmp))) \n",
    "        } else {\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- pre_tmp_IS$EndMotif_tmp\n",
    "        }\n",
    "        \n",
    "        tmp_IS_meta <- merge(x = samples_metadata, y = tmp_IS, by.x = \"sample_id\", by.y = \"sample_id\") \n",
    "        if (!exists(\"Asp_IS_META\")){Asp_IS_META <- tmp_IS_meta} else {\n",
    "            Asp_IS_META <- rbind(Asp_IS_META, tmp_IS_meta)}\n",
    "    }\n",
    "    if (exists(\"Asp_IS_META\")){\n",
    "        fwrite(Asp_IS_META, \"../../output/PP_figures/Asp_IS_meta.csv\", row.names = FALSE)\n",
    "    }\n",
    "}\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (exists(\"unmappedHost_IS_META\")){rm(\"unmappedHost_IS_META\")}\n",
    "profvis({\n",
    "for (file in list.files(DIR_IN, pattern = \"___.*_R(1|2)(pos|neg)_TLEN_EndMotif.txt\")){\n",
    "    sample_tmp <- unlist(strsplit(file, \"_\"))[1] \n",
    "    taxID_tmp <- unlist(strsplit(file, \"_\"))[4]\n",
    "    mapGenome_tmp <- unlist(strsplit(file, \"_\"))[5]\n",
    "    side_tmp <- unlist(strsplit(file, \"_\"))[6]\n",
    "    \n",
    "    if(countLines(paste0(DIR_IN, file)) > 1){\n",
    "        pre_tmp_IS <- fread(paste0(DIR_IN, file), sep = \",\") %>% \n",
    "                    dplyr::rename(EndMotif_tmp = EndMotif) %>% \n",
    "                    mutate(sample_id = sample_tmp) %>% \n",
    "                    mutate(side = side_tmp) %>% \n",
    "                    mutate(taxID = taxID_tmp) %>% \n",
    "                    mutate(mapGenome = mapGenome_tmp) %>% \n",
    "                    mutate(read = substr(side_tmp, 0, 2)) %>% \n",
    "                    filter(TLEN != 0) %>% #remove TLEN not equal to 0\n",
    "                    filter(abs(TLEN) < 1000) %>% #filter max TLEN of 1000\n",
    "                    filter(!grepl(EndMotif_tmp, pattern = \"N\")) %>% #remove reads with end-motif including \"N\"\n",
    "                    group_by(TLEN, EndMotif_tmp, sample_id, side, taxID, mapGenome) %>% \n",
    "                    summarise(Count = sum(Count), .groups = \"keep\")\n",
    "        \n",
    "        if (grepl(side_tmp, pattern = \"neg\")){\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- as.character(reverseComplement(DNAStringSet(pre_tmp_IS$EndMotif_tmp))) \n",
    "        } else {\n",
    "            tmp_IS <- pre_tmp_IS \n",
    "            tmp_IS$EndMotif <- pre_tmp_IS$EndMotif_tmp\n",
    "        }\n",
    "        \n",
    "        tmp_IS_meta <- merge(x = samples_metadata, y = tmp_IS, by.x = \"sample_id\", by.y = \"sample_id\") \n",
    "        if (!exists(\"unmappedHost_IS_META\")){unmappedHost_IS_META <- tmp_IS_meta} else {\n",
    "            unmappedHost_IS_META <- rbind(unmappedHost_IS_META, tmp_IS_meta)}\n",
    "    }\n",
    "    if (exists(\"unmappedHost_IS_META\")){\n",
    "        fwrite(unmappedHost_IS_META, \"../../output/PP_figures/unmappedHost_IS_meta.csv\", row.names = FALSE)\n",
    "    }\n",
    "}\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Asp_IS_META$taxID %>% unique()\n",
    "Asp_IS_META %>% \n",
    "    mutate(read = substr(side, 1,2)) %>% \n",
    "    group_by(sample_id, sample_type, taxID, read, mapGenome) %>% \n",
    "    summarise(sumCount = sum(Count), .groups = \"keep\") %>% \n",
    "    filter(read == \"R1\") %>% \n",
    "    mutate(taxID = \n",
    "           ifelse(taxID == \"1131492\", yes = \"k2_Aspergillaceae\", no = \n",
    "           ifelse(taxID == \"5052\", yes = \"k2_Aspergillus\", no = \n",
    "           ifelse(taxID == \"746128\", yes = \"k2_Aspergillus_fumigatus\", no = \n",
    "           ifelse(taxID == \"\", yes = \"unmappedHost\", no = taxID))))) %>% \n",
    "    mutate(taxID = ifelse(taxID == \"\", yes = \"unmappedHost\", no = taxID)) %>% \n",
    "    reshape2::dcast(formula = sample_id + sample_type + read + mapGenome ~ taxID, value.var = \"sumCount\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Asp_IS_META %>% \n",
    "    mutate(read = substr(side, 1,2)) %>% \n",
    "    mutate(TLEN = abs(TLEN)) %>% \n",
    "    group_by(sample_id, sample_type, taxID, read, mapGenome, TLEN) %>% \n",
    "    summarise(sumCount = sum(Count), .groups = \"keep\") %>% \n",
    "    filter(read == \"R1\") %>% \n",
    "    filter(mapGenome == \"AfumigatusA1163\" ) %>%\n",
    "    mutate(taxID = \n",
    "           ifelse(taxID == \"1131492\", yes = \"k2_Aspergillaceae\", no = \n",
    "           ifelse(taxID == \"5052\", yes = \"k2_Aspergillus\", no = \n",
    "           ifelse(taxID == \"746128\", yes = \"k2_Aspergillus_fumigatus\", no = \n",
    "           ifelse(taxID == \"\", yes = \"unmappedHost\", no = taxID))))) %>% \n",
    "    ggplot(aes(x = TLEN, y = sumCount, group = taxID, col = taxID)) + \n",
    "        geom_segment(aes(x = TLEN, xend = TLEN, y = 0, yend = sumCount)) + \n",
    "        facet_grid(rows = vars(taxID), cols = vars(sample_id)) + theme_bw()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:r4.2.0]",
   "language": "R",
   "name": "conda-env-r4.2.0-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
